includes/wc-template-functions.php

```diff
@@ -494,10 +494,8 @@ function wc_product_post_class( $classes, $class = '', $post_id = 0 ) {
 		if ( $product->get_type() ) {
 			$classes[] = 'product-type-' . $product->get_type();
 		}
-		if ( $product->is_type( 'variable' ) ) {
-			if ( ! $product->get_default_attributes() ) {
-				$classes[] = 'has-default-attributes';
-			}
+		if ( $product->is_type( 'variable' ) && $product->get_default_attributes() ) {
+			$classes[] = 'has-default-attributes';
```

Comment:
This logic has changed. I assume the old version was a bug. Just pointing it out, in case the change was unintentional. :)

Reply:
Yup, it was a bug :)